genie-web/src/main/java/com/netflix/genie/web/services/AgentFilterService.java

```diff
@@ -0,0 +1,87 @@
+/*
+ *
+ *  Copyright 2019 Netflix, Inc.
+ *
+ *     Licensed under the Apache License, Version 2.0 (the "License");
+ *     you may not use this file except in compliance with the License.
+ *     You may obtain a copy of the License at
+ *
+ *         http://www.apache.org/licenses/LICENSE-2.0
+ *
+ *     Unless required by applicable law or agreed to in writing, software
+ *     distributed under the License is distributed on an "AS IS" BASIS,
+ *     WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ *     See the License for the specific language governing permissions and
+ *     limitations under the License.
+ *
+ */
+
+package com.netflix.genie.web.services;
+
+import com.netflix.genie.common.internal.dto.v4.AgentClientMetadata;
+import com.netflix.genie.common.internal.exceptions.unchecked.GenieAgentRejectedException;
+import lombok.AllArgsConstructor;
+import lombok.Getter;
+
+import javax.validation.Valid;
+
+/**
+ * Service to identify agent/clients that the server wants to refuse service to.
+ * To be used, for example, to blacklist clients running a given version that is known to be deprecated or otherwise
+ * incompatible.
+ *
+ * @author mprimi
+ * @since 4.0.0
+ */
+public interface AgentFilterService {
```

Comment:
Would `AgentVerificationService` be a better name? Just want to make clear what it is. I guess HTTP Filters can do rejections too so I could see why this would work as well just throwing it out there